great_expectations/rule_based_profiler/data_assistant_result/data_assistant_result.py

```diff
@@ -3156,104 +3390,136 @@ def _filter(domain: Domain) -> bool:
-        ordinal_metrics: Set[str] = self._get_metric_types_from_altair_type(
+        ordinal_metrics: Set[str] = self._get_sanitized_metric_names_from_altair_type(
             altair_type=AltairDataTypes.ORDINAL
         )
-        quantitative_metrics: Set[str] = self._get_metric_types_from_altair_type(
+        quantitative_metrics: Set[
+            str
+        ] = self._get_sanitized_metric_names_from_altair_type(
             altair_type=AltairDataTypes.QUANTITATIVE
         )
-        temporal_metrics: Set[str] = self._get_metric_types_from_altair_type(
+        temporal_metrics: Set[str] = self._get_sanitized_metric_names_from_altair_type(
             altair_type=AltairDataTypes.ORDINAL
         )
 
-        if plot_mode is PlotMode.PRESCRIPTIVE:
-            plot_impl: Optional[
+        if plot_mode is PlotMode.DIAGNOSTIC:
+            expectation_plot_impl: Optional[
                 Callable[
                     [
                         str,
                         pd.DataFrame,
-                        str,
+                        Set[str],
                         bool,
                         Optional[str],
                     ],
-                    alt.Chart,
+                    Union[alt.Chart, alt.LayerChart],
                 ]
             ] = None
 
-            if metric_name in nominal_metrics:
-                plot_impl = self._get_expect_domain_values_to_match_set_chart
-            elif metric_name in ordinal_metrics:
-                plot_impl = self._get_expect_domain_values_ordinal_chart
-            elif metric_name in quantitative_metrics:
-                plot_impl = self._get_expect_domain_values_to_be_between_chart
-            elif metric_name in temporal_metrics:
-                plot_impl = self._get_expect_domain_values_temporal_chart
+            if DataAssistantResult._all_metric_names_in_iterable(
+                metric_names=sanitized_metric_names, iterable=nominal_metrics
+            ):
+                expectation_plot_impl = (
+                    self._get_expect_domain_values_to_match_set_chart
+                )
+            elif DataAssistantResult._all_metric_names_in_iterable(
+                metric_names=sanitized_metric_names, iterable=ordinal_metrics
+            ):
+                expectation_plot_impl = self._get_expect_domain_values_ordinal_chart
+            elif DataAssistantResult._all_metric_names_in_iterable(
+                metric_names=sanitized_metric_names, iterable=quantitative_metrics
+            ):
+                expectation_plot_impl = (
+                    self._get_expect_domain_values_to_be_between_chart
+                )
+            elif DataAssistantResult._all_metric_names_in_iterable(
+                metric_names=sanitized_metric_names, iterable=temporal_metrics
+            ):
+                expectation_plot_impl = self._get_expect_domain_values_temporal_chart
 
-            return plot_impl(
+            return expectation_plot_impl(
```

Comment:
expectation_plot_impl could be `None` here since it starts that way and all these `elif`s could evaluate to false.

Reply:
🙇🏻